src/Core/Injector/InjectionCreator.php

```diff
@@ -2,29 +2,16 @@
 
 namespace SilverStripe\Core\Injector;
 
-use ReflectionClass;
-use ReflectionException;
-
 /**
  * A class for creating new objects by the injector.
  */
 class InjectionCreator implements Factory
 {
-
     public function create($class, array $params = [])
     {
-        try {
-            $reflector = new ReflectionClass($class);
-        } catch (ReflectionException $e) {
-            throw new InjectorNotFoundException($e);
+        if (!class_exists($class)) {
+            throw new InjectorNotFoundException("Class {$class} does not exist");
         }
-
-        if (count($params)) {
-            // Remove named keys to ensure that PHP7 and PHP8 interpret these the same way
-            $params = array_values($params);
-            return $reflector->newInstanceArgs($params);
-        }
-
-        return $reflector->newInstance();
+        return new $class(...array_values($params));
```

Comment:
Is it necessary to call `array_values` if you're using the spread (`...`) operator? I would have expected that to ignore keys.

If it is necessary, might be worth keeping the comment about keeping things consistent between php7 and 8.

Reply:
yes

```
~ $ php -a
Interactive mode enabled

php > function f($a,$b){print_r([$a,$b]);};
php > f(...[1,2]);
Array
(
    [0] => 1
    [1] => 2
)
php > f(...['a'=>3,'b'=>4]);
PHP Warning:  Uncaught Error: Cannot unpack array with string keys in php shell code:1
Stack trace:
#0 {main}
  thrown in php shell code on line 1
```

Reply:
will add back inline comment to explain array values

Reply:
The spread operator works with string keys from PHP 8.1, older versions need to use the array_values approach.